interfaces/system_key.go

```diff
@@ -378,11 +384,68 @@ func RemoveSystemKey() error {
 }
 
 func MockSystemKey(s string) func() {
-	var sk systemKey
-	err := json.Unmarshal([]byte(s), &sk)
+	sk, err := SystemKeyFromString(s)
 	if err != nil {
 		panic(err)
 	}
-	mockedSystemKey = &sk
+	mockedSystemKey = sk.(*systemKey)
 	return func() { mockedSystemKey = nil }
 }
+
+type SystemKeyMismatchAction string
+
+const (
+	SystemKeyMismatchProceed            = SystemKeyMismatchAction("proceed")
+	SystemKeyMismatchRegenerateProfiles = SystemKeyMismatchAction("regenerate-profiles")
+)
```

Comment:
are we going to serialize this somewhere? why not use int for these here?

Reply:
yeah, it's logged, but I'll make it an int and implement a Stringer()